web/packages/design/src/SlideTabs/SlideTabs.tsx

```diff
@@ -16,65 +16,117 @@
+    }
+  }, [activeIndex]);
+
+  function handleKeyDown(e: React.KeyboardEvent) {
+    if (e.key === 'ArrowRight' && activeIndex < tabs.length - 1) {
+      onChange(activeIndex + 1);
+    }
+    if (e.key === 'ArrowLeft' && activeIndex > 0) {
+      onChange(activeIndex - 1);
+    }
+  }
+
+  // The component structure was designed according to
+  // https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-automatic/.
   return (
-    <Wrapper>
-      <TabNav role="tablist" appearance={appearance} size={size}>
-        {tabs.map((tabName, tabIndex) => {
+    // A container that displays background and sets up padding for the slider
+    // area. It's separate from the tab list itself, since we need to
+    // absolutely position the slider relative to this container's content box,
+    // and not its padding box. So we set up padding if needed on this one, and
+    // then position the slider against the tab list.
+    <Wrapper fitContent={fitContent} size={size} appearance={appearance}>
+      <TabList
+        ref={tabContainer}
+        role="tablist"
+        itemCount={tabs.length}
+        onKeyDown={handleKeyDown}
+      >
+        {tabs.map((tabSpec, tabIndex) => {
           const selected = tabIndex === activeIndex;
-          let onClick;
+          const tabName =
+            typeof tabSpec === 'string' ? tabSpec : tabSpec.content;
+          const controls =
+            typeof tabSpec === 'string' ? undefined : tabSpec.controls;
+          const tabId = typeof tabSpec === 'string' ? tabIndex : tabSpec.id;
+
+          let onClick = undefined;
           if (!disabled && !isProcessing) {
-            onClick = (e: React.MouseEvent<HTMLLabelElement, MouseEvent>) => {
+            onClick = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
               e.preventDefault();
               onChange(tabIndex);
             };
           }
+
           return (
-            <TabLabel
+            <TabButton
+              key={tabId}
+              ref={selected ? activeTab : undefined}
               role="tab"
-              htmlFor={`${name}-${tabName}`}
               onClick={onClick}
-              itemCount={tabs.length}
-              key={`${tabName}-${tabIndex}`}
+              selected={selected}
               className={selected ? 'selected' : undefined}
+              aria-controls={controls}
+              tabIndex={!disabled && selected ? 0 : -1}
               processing={isProcessing}
               disabled={disabled}
               aria-selected={selected}
+              size={size}
             >
-              <Box>
+              {/* We need a separate tab content component, since the spinner,
+                  when displayed, shouldn't take up space to prevent layout
+                  jumping. TabContent serves as a positioning anchor whose left
+                  edge is the left edge of the content (not the tab button,
+                  which can be much wider). */}
+              <TabContent>
```

Comment:
I see the problem, but I'm not sure if displaying the spinner with `position: absolute` helps.
If there is not enough space, the spinner covers other elements:
<img width="374" alt="image" src="https://github.com/user-attachments/assets/f4283740-c105-4e3f-ad97-5225fa9b122a">

Maybe instead of using a spinner we should apply an animation, for example, similar to the one that we have in the unified view when it is loading?

Reply:
Oh, wait. I think I'll probably also need to display a status icon in the "wider" UI version, too. Later, I may need another slot to the tab spec to add a temporary icon decoration, and this one will occupy the same place as the spinner does currently.

In general, though, I don't want to invest too much time to make it perfect, since the UX team also works on a redesign of this component, and although I drawn from the redesign, it's not finalized, so if I spend too much time on it, it's most certainly gonna be wasted.

Reply:
Sure, I thought we have a final design. So let's keep the spinner as is, this PR doesn't touch it actually.